### PR TITLE
Normalize queries on workspaces

### PR DESCRIPTION
**Pull request by samuelechu**
This pr makes workspace queries always have the same form, instead of being possibly transformed into a executable query. Doing this will aid in not having to transform these queries when doing create or save mutations.

### CODE REVIEW

File: src/main/webapp/components/workspace/workspace.js [hunk 1]
```diff
@@ -74,6 +94,23 @@ export default () => {
   const [queries, setQueries] = useState()
   const { results, status, onSearch, onCancel, onClear } = useQueryExecutor()
 
+  const [queuedSearch, setQueuedSearch] = useState('')
+
+  React.useEffect(
```
Comment (samuelechu) [inline src/main/webapp/components/workspace/workspace.js]: This logic will hopefully be taken out in a future pr

File: src/main/webapp/components/query-manager/query-manager.stories.tsx [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { button } from '@storybook/addon-knobs'
 
 stories.add('Basic', () => {
   const [queries, setQueries] = useState(sampleQueries)
-  const [query, setQuery] = useState(sampleQueries[0].id)
+  const [query, setQuery] = React.useState(sampleQueries[0].id)
```
Comment (willwill96) [inline src/main/webapp/components/query-manager/query-manager.stories.tsx]: :pencil2: unnecessary change

File: src/main/webapp/components/query-manager/query-manager.tsx [hunk 3]
```diff
@@ -64,19 +64,25 @@ const QueryCard = (props: QueryCardProps) => {
...
   const hasQueries = queries && queries.length > 0
   const [anchorEl, handleOpen, handleClose, open] = useAnchorEl()
 
   const queryCards = queries.map(query => (
-    <QueryCard {...props} query={query} key={query.id} />
+    <QueryCard
+      {...props}
+      query={query}
+      key={query.id}
+      onSearch={() => onSearch(query.id!)}
```
Comment (willwill96) [inline src/main/webapp/components/query-manager/query-manager.tsx]: just curious, what does the syntax of `query.id!` do specifically the `!`?

File: src/main/webapp/components/query-manager/query-manager.stories.tsx (hunk 2 above)
Comment (samuelechu) [reply src/main/webapp/components/query-manager/query-manager.stories.tsx]: I didn't want this action to trigger when selecting a query https://github.com/connexta/revelio/blob/master/src/main/webapp/@storybook/use-state.js#L8

File: src/main/webapp/components/query-manager/query-manager.tsx (hunk 3 above)
Comment (samuelechu) [reply src/main/webapp/components/query-manager/query-manager.tsx]: I think it assures the compiler that the optional won't be undefined. It was complaining before I added the `!` https://stackoverflow.com/a/42274019
